src/GitHub.VisualStudio/UI/Views/GitHubPaneViewModel.cs

```diff
@@ -105,53 +108,103 @@ protected override void RepoChanged(bool changed)
             if (!changed)
                 return;
 
+            Stop();
             IsGitHubRepo = null;
             Reload().Forget();
         }
 
+        /// <summary>
+        /// This method is reentrant, so all await calls need to be done before
+        /// any actions are performed on the data. More recent calls to this method
+        /// will cause previous calls pending on await calls to exit early.
+        /// </summary>
+        /// <returns></returns>
         async Task Reload([AllowNull] ViewWithData data = null, bool navigating = false)
         {
+            if (!initialized)
+                return;
+
+            latestReloadCallId++;
+            var reloadCallId = latestReloadCallId;
+
             navigatingViaArrows = navigating;
 
             if (!IsGitHubRepo.HasValue)
-                IsGitHubRepo = await IsAGitHubRepo();
-
-            if (!IsGitHubRepo.Value)
             {
-                if (uiController != null)
-                {
-                    Stop();
-                    //var factory = ServiceProvider.GetExportedValue<IUIFactory>();
-                    //var c = factory.CreateViewAndViewModel(UIViewType.LoggedOut);
-                    //Control = c.View;
-                }
-                return;
+                var isGitHubRepo = await IsAGitHubRepo();
+                if (reloadCallId != latestReloadCallId)
+                    return;
+
+                IsGitHubRepo = isGitHubRepo;
             }
 
             var connection = await connectionManager.LookupConnection(ActiveRepo);
-            IsLoggedIn = await connection.IsLoggedIn(hosts);
+            if (reloadCallId != latestReloadCallId)
+                return;
 
-            if (IsLoggedIn)
+            if (connection == null)
+                IsLoggedIn = false;
+            else
             {
-                if (uiController == null || (data != null && data.ActiveFlow != uiController.SelectedFlow))
-                    StartFlow(data?.ActiveFlow ?? UIControllerFlow.PullRequests, connection, data);
-                else if (data != null || currentNavItem >= 0)
-                    uiController.Jump(data ?? navStack[currentNavItem]);
+                var isLoggedIn = await connection.IsLoggedIn(hosts);
+                if (reloadCallId != latestReloadCallId)
+                    return;
+
+                IsLoggedIn = isLoggedIn;
             }
+
+            if (!IsGitHubRepo.Value)
+            {
+                //LoadView(UIViewType.NotAGitHubRepo);
```

Comment:
Doesn't look like this if statement is doing anything anymore, I uncommented to investigate and it looks like it's broken, do we intend to add `NotAGitHubRepo` to `UIViewType` at some point in the future?